com.unity.ml-agents/Runtime/Sensors/StackingSensor.cs

```diff
@@ -85,11 +99,32 @@ public int Write(ObservationWriter writer)
 
             // Now write the saved observations (oldest first)
             var numWritten = 0;
-            for (var i = 0; i < m_NumStackedObservations; i++)
+            if (wrappedShape.Length == 1)
```

Comment:
Found this bug that I didn't change the Write method properly for uncompressed obs and it's still stacking on the first dimension. Here changed it to the last dimension for 2D observations so that inference can work properly.

I left 1D unchanged (stacking on first dim) since it has to be handled differently than 2D anyway and otherwise all models have to be re-trained. Thoughts?